src/factor_graph.h

```diff
@@ -174,16 +174,13 @@ class CompactFactorGraph {
 
 inline double CompactFactorGraph::potential(
     const CompactFactor& factor, const variable_value_t assignments[],
-    factor_value_t factor_value) {
-  // For boolean, this is stored in CompactFactor.value;
-  //     caller shouldn't set the factor_value arg.
-  // For categorical, this is stored in Factor.factor_params.value.
-  //     caller should pass in via the factor_value arg.
+    feature_value_t feature_value) {
+  // For boolean, feature_value is stored in CompactFactor.value;
+  //     caller shouldn't set the feature_value arg.
+  // For categorical, feature_value is stored in Factor.factor_params.value.
+  //     caller should pass in via the feature_value arg.
   // TODO: better data structures than the factor_params duct tape...
-  if (factor_value == Factor::DEFAULT_VALUE) {
-    factor_value = factor.value;
-  }
-  return factor.potential(vifs.get(), assignments, -1, -1) * factor_value;
+  return factor.potential(vifs.get(), assignments, -1, -1, feature_value);
```

Comment:
I guess we could just get rid of this wrapper since it's confusing and doesn't add much value.  Seems like used from only one place.. `update_weight`